src/htm/algorithms/Connections.cpp

```diff
@@ -479,6 +475,9 @@ void Connections::adaptSegment(const Segment segment,
     }
   }
 
+  //balance synapses using competition on dendrite
+  synapseCompetition(segment, 4, 10); //FIXME derive these numbers
```

Comment:
calling `synapseCompetition` from `adaptSegment`
- [x] this works pretty well, with synapse pruning many synapses get removed which leads to a considerable speedup
- [ ] set the min/max parameters somehow smarter
   - I guess those should ideally be around stimulusThreshold ?
   - the tighter the boundary, the more syns get removed, the faster SP (on MNIST) is. 
     - but where is the line? What is the disadvantage? I'd think too tight boundary is more prone to noice?

CC @ctrl-z-9000-times